sdk/granter.go

```diff
@@ -228,21 +228,52 @@ func NewGranterFromService(ctx context.Context, as attributes.AttributesServiceC
 		def := pair.GetAttribute()
 		if def != nil {
 			grants.addAllGrants(fqn, def.GetGrants(), def)
+			storeKeysToCache(def.GetGrants(), keyCache)
 		}
 		v := pair.GetValue()
 		if v != nil {
 			grants.addAllGrants(fqn, v.GetGrants(), def)
+			storeKeysToCache(v.GetGrants(), keyCache)
 		}
 	}
 
 	return grants, nil
 }
 
+func algProto2String(e policy.KasPublicKeyAlgEnum) string {
+	switch e {
+	case policy.KasPublicKeyAlgEnum_KAS_PUBLIC_KEY_ALG_ENUM_EC_SECP256R1:
+		return "ec:secp256r1"
+	case policy.KasPublicKeyAlgEnum_KAS_PUBLIC_KEY_ALG_ENUM_RSA_2048:
+		return "rsa:2048"
+	case policy.KasPublicKeyAlgEnum_KAS_PUBLIC_KEY_ALG_ENUM_UNSPECIFIED:
+		return ""
+	}
+	return ""
+}
+
+func storeKeysToCache(kases []*policy.KeyAccessServer, c *kasKeyCache) {
+	for _, kas := range kases {
+		if kas.GetPublicKey() == nil || kas.GetPublicKey().GetCached() == nil || kas.GetPublicKey().GetCached().GetKeys() == nil || len(kas.GetPublicKey().GetCached().GetKeys()) == 0 {
```

Comment:
The proto accessors should prevent nil panics here, so you could go all the way to `.GetKeys()` and just check nil/length there?

Reply:
yes, you are correct. this explains why the linter loves requiring the getter